TestUtilities/Mocks/SingleFeatureCoreMock.swift

```diff
@@ -0,0 +1,78 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+import XCTest
+import DatadogInternal
+
+/// Single-Feature core mocks feature-scope for a single `Feature` type.
+public final class SingleFeatureCoreMock<Feature>: DatadogCoreProtocol, FeatureScope where Feature: DatadogFeature {
```

Comment:
nit/ Not very clear when to use this instead of `PassthroughCoreMock`. As we're adding this into shared `TestUtilities`, IMO we should improve this comment.

Reply:
Indeed, I'm also making this a subclass of `PassthroughCoreMock`.